### DISCUSSION TITLE
Input requested: considering requiring --ttl on all bkt invocations

### DISCUSSION THREAD

**Discussion by dimo414** [dimo414/bkt#48] [Polls]
I am considering removing the [default value of one minute](https://github.com/dimo414/bkt/blob/84c483a241aa6a0c72e5d447963eab19d8e14576/src/main.rs#L139C42-L139C63) from `--ttl`. This would be a breaking change to the CLI, which I generally want to avoid, so I'm soliciting input before making such a change.

This default is fairly arbitrary (it's a holdover from [bash-cache](https://github.com/dimo414/bash-cache/blob/b407d336b8476453483ce66b5df3c7efb073b7d6/bash-cache.sh#L233) which already discourages relying on the default); there's no reason to think a given invocation _should_ use this value by default. The primary justification for keeping it is to have minimal caching be as simple as prepending `bkt --` to a command, however I worry this is misleading as both callers and readers actually do need to understand the TTL semantics in order to understand what this does. `bkt -- date; sleep 59; bkt -- date` and `bkt -- date; sleep 61; bkt -- date` behave differently and a casual reading of these calls cannot discern this.

My real motivation, aside from the default value being poor, is related to #27 and potentially support caching _without_ a TTL if a modtime file is specified instead. It's possible to implement #27 without removing this default (and specifying both a TTL and a modtime file will always be valid) but it would be a stronger feature if it could eliminate needing to specify a TTL, so I'd like to remove the TTL default before taking a swing at that feature.

My plan would be to cut a new minor version (it would be a major version bump but `bkt` is still a `0.*.*` project) with _just_ the removal of this default. Hopefully that will help existing usages relying on the default migrate safely.

Note the legacy behavior can be preserved trivially by adding `export BKT_TTL=60s` to your `.bashrc` (or setting the same variable in whichever environment needs it).

**Comment by dimo414** [comment]
I'm planning to make this change in 0.8.0, which will be cut soon.

**Comment by borestad** [comment]
Some food for thought. If bkt had a config file i.e `$HOME/.bkt.yaml` , this could be set by default in userland instead.

**Reply by dimo414** [reply]
True, but you could accomplish the same effect by setting the env var in your `~/.bashrc` or similar. I'd be hesitant to make something like the TTL a globally-configured option for the same reasons I want to rip out the default in the first place - it really doesn't make much sense to have one.

But thanks for the suggestion - if you have particular use-cases for a `.bkt` config file of some sort feel free to start a discussion or open an FR!
